psi4/src/psi4/libfock/snLinK.cc

```diff
@@ -0,0 +1,548 @@
+        }
+    }
+   
+    // we are done
+    return permutation_matrix;
+}
+
+// converts a Psi4::Molecule object to a GauXC::Molecule object
+GauXC::Molecule snLinK::psi4_to_gauxc_molecule(std::shared_ptr<Molecule> psi4_molecule) {
+    GauXC::Molecule gauxc_molecule;
+
+    for (size_t iatom = 0; iatom != psi4_molecule->natom(); ++iatom) {
+        auto atomic_number = psi4_molecule->true_atomic_number(iatom);
+        auto x_coord = psi4_molecule->x(iatom);
+        auto y_coord = psi4_molecule->y(iatom);
+        auto z_coord = psi4_molecule->z(iatom);
+        
+        gauxc_molecule.emplace_back(GauXC::AtomicNumber(atomic_number), x_coord, y_coord, z_coord);
+    }
+
+    return gauxc_molecule;
+}
+
+// converts a Psi4::BasisSet object to a GauXC::BasisSet object
+template <typename T>
+GauXC::BasisSet<T> snLinK::psi4_to_gauxc_basisset(std::shared_ptr<BasisSet> psi4_basisset, double basis_tol, bool force_cartesian) {
+    using prim_array = typename GauXC::Shell<T>::prim_array;
+    using cart_array = typename GauXC::Shell<T>::cart_array;
+
+    GauXC::BasisSet<T> gauxc_basisset(psi4_basisset->nshell());
+ 
+    for (size_t ishell = 0; ishell != psi4_basisset->nshell(); ++ishell) {
+        auto psi4_shell = psi4_basisset->shell(ishell);
+       
+        const auto nprim = GauXC::PrimSize(psi4_shell.nprimitive());
+        prim_array alpha; 
+        prim_array coeff;
+
+        for (size_t iprim = 0; iprim != psi4_shell.nprimitive(); ++iprim) {
+            alpha.at(iprim) = psi4_shell.exp(iprim);
+            coeff.at(iprim) = psi4_shell.coef(iprim);
+        }
+
+        auto psi4_shell_center = psi4_shell.center();
+        cart_array center = { psi4_shell_center[0], psi4_shell_center[1], psi4_shell_center[2] };
+
+        gauxc_basisset[ishell] = GauXC::Shell(
+            nprim,
+            GauXC::AngularMomentum(psi4_shell.am()), 
+            (force_cartesian ? GauXC::SphericalType(false) : GauXC::SphericalType( !(psi4_shell.is_cartesian()) ) ),
+            alpha,
+            coeff,
+            center,
+            false // do not normalize shell via GauXC; it is normalized via Psi4
+        );
+    }
+    
+    for (auto& sh : gauxc_basisset) {
+        sh.set_shell_tolerance(basis_tol); 
+    }
+
+    return gauxc_basisset;
+}
+
+#endif 
+
+// ==> SplitJK-inherited functions go here <== //
+
+snLinK::snLinK(std::shared_ptr<BasisSet> primary, Options& options) : SplitJK(primary, options) {
+
+#ifdef USING_gauxc
+    timer_on("snLinK: Setup");
+    
+    // => Part #1: Options Processing <= //
+    timer_on("snLinK: Options Processing");
+
+    // set a few Psi4-specific parameters
+    incfock_iter_ = false;
+  
+    basis_tol_ = options.get_double("SNLINK_BASIS_TOLERANCE");
+
+    pruning_scheme_ = options_.get_str("SNLINK_PRUNING_SCHEME");
+    radial_scheme_ = options_.get_str("SNLINK_RADIAL_SCHEME");
+ 
+//#if psi4_SHGSHELL_ORDERING == LIBINT_SHGSHELL_ORDERING_STANDARD
+#if psi4_SHGSHELL_ORDERING == 1 
```

Comment:
Previously, it was indeed working differently; the use of `LIBINT_SHGSHELL_ORDERING_*` would throw a compiler error since the macros were not recognized in `snLinK.cc`. Switching it over now, however, it seems to work for whatever reason, so I'll use the `LIBINT_SHGSHELL_ORDERING_*` notation.